lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -288,30 +266,171 @@ def create_instance(instance_params):
             enable_automatic_recovery=instance_params['auto_recovery_enabled'],
             description=instance_params['description'])
 
-        result['api_request_body'] = str(body)
+        if module._verbosity >= 3:
+            result['api_request_body'] = str(body)
         try:
             # Create an application from a template
             api_response = api_instance.create_application_from_template_using_post(
                 body)
-            result['changed'] = True
-            result['failed'] = False
-            module.exit_json(**result)
-
+            if module._verbosity >= 3:
+                result['api_response'] = str(api_response)
         except ApiException as e:
 
             return "Exception when calling ApplicationsApi->create_application_from_template_using_post: %s\n" % e
 
-    create_instance_response = create_instance(instance_params)
+        tacp_utils.wait_for_action_to_complete(
+            api_response.action_uuid, api_client)
+        result['ansible_module_results'] = tacp_utils.get_resource_by_uuid(
+            api_response.object_uuid, 'application', api_client)
+        result['changed'] = True
+
+    def instance_power_action(name, api_client, action):
+        instance_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['name'], 'application', api_client)
+        api_instance = tacp.ApplicationsApi(api_client)
+
+        actions = {"start": api_instance.start_application_using_put,
+                   "stop": api_instance.stop_application_using_put,
+                   "restart": api_instance.restart_application_using_put,
+                   "shutdown": api_instance.shutdown_application_using_put,
+                   "pause": api_instance.pause_application_using_put,
+                   "resume": api_instance.resume_application_using_put,
```

Comment:
Thumbs up for verbosity in such cases as `resume` vs `put`

Some constants would be nice not to hardcode strings around here and there. Think choices as in:

`Action.START = 'start'` 

Then instead of hardcoding `start` string all over the code just use `Action.START` constant

Reply:
Sounds good, I think I should include this in the restructuring branch I'm starting on now - are you good to delay this from this PR?

Reply:
Sure. I would have requested changes otherwise